lib/internal/serve.js

```diff
@@ -0,0 +1,289 @@
+"use strict";
+
+// Adapted from wpt tools
+// https://github.com/web-platform-tests/wpt/blob/926d722bfc83f3135aab36fddc977de82ed7e63e/tools/serve/serve.py
+
+const assert = require("assert");
+const fs = require("fs");
+const { parse: parseUrl } = require("url");
+const { filesystemPath } = require("./handlers.js");
+const { jsMetaRegexp, parseVariants, readScriptMetadata, replaceEnd } = require("./sourcefile.js");
+
+/**
+ * @abstract
+ */
+class WrapperHandler {
+  constructor(basePath, urlBase = "/") {
+    this.basePath = basePath;
+    this.urlBase = urlBase;
+  }
+
+  handleRequest(request, response) {
+    for (const [headerName, headerValue] of this.headers()) {
+      response.setHeader(headerName, headerValue);
+    }
+
+    this.checkExposure(request);
```

Comment:
Oh, shoot, I still need to add a `try..catch` here to turn the exceptions into error responses. I'll do that later today.